docs/proposals/improved-schema-for-services.md

```diff
@@ -0,0 +1,442 @@
+
+# Improved schema for operator backed services
+
+This proposal is about improving the experience for a user of the operator backed services. Currently a user doesn't know exaustively which fields are available in a CR as we depend on optional metadata present in CRDs, to improve this we are considering using the metadata available from the cluster about the CRDs.
+Getting that metadata from the cluster is challenging because a normal cluster user ( plain vanilla ) doesn't have access to that metadata so we are gonna follow a 3 tiered approach
+
+- check if the user has access to the CRD, if so then use that.
+- get the `swagger.json` from the openapi endpoint provided by kubernetes. a sample can be found here https://gist.github.com/girishramnani/f29949d5cb8c6547102776437e05ac19
+- finally if we cannot get the openapi data as well then use the `ClusterServiceVersion` to get the parameters.
+
+This change would affect multiple service commands and the changes are described briefly below -
+- `odo catalog describe service` should include the metadata fields with description so the users can provide then when doing `odo service create`. 
+- `odo catalog list service` shouldn't change much other then listing CRDs per operator
+- `odo service create` should take flags and dynamically fill the CRD structs - metadata would be used for validation
+- `odo service create --from-file` would be used by adapter team or whoever wants to provide the whole CRD themselves as a file. this feature is already present and working.
+- `json` output for all commands.
+
+# Implementation plan
+
+Note - Below I would mention `cluster` manytimes and that means both openshift and kuberenetes. If there is something specific to one or the other then I would mention it.
+
+Below is the step wise plan for implementing this feature - 
+
+## Getting the metadata
+
+### Admin Access/Access to CRD API path
+First we would consider the scenerio where the user has admin access or privileges to access the CRD API to the cluster.
```

Comment:
Just to be on the same page. When this says "**First**", It means in code flow/logic. 
As far as the implementation plan goes it will be actually done in reverse order. 
First, the `ClusterServiceVersion` approach will be implemented (already in progress)
Than Swagger approach
Then the "admin" approach will be implemented last.

Reply:
yes

Reply:
made it clear in the proposal by adding an explicit order of implementation and changed the working at places to point out that "above steps might not be implemented yet"